climate-monitoring-client/src/main/java/climatemonitoring/Registration.java

```diff
@@ -1,18 +1,154 @@
 package climatemonitoring;
 
+import climatemonitoring.core.ConnectionLostException;
+import climatemonitoring.core.DatabaseRequestException;
+import climatemonitoring.core.Operator;
 import climatemonitoring.core.ViewState;
+import climatemonitoring.core.headless.Console;
+import climatemonitoring.core.utility.Email;
+
+import java.util.Random;
 
 class Registration extends ViewState {
 
 	@Override
 	public void onHeadlessRender(String args) {
 
-		throw new UnsupportedOperationException("Unimplemented method 'onHeadlessRender'");
+		String errorMsg = null;
+
+		do {
+
+			m_userID = Console.read("User ID > ");
+			errorMsg = Check.userID(m_userID);
+
+			if (errorMsg != null)
+				Console.write(errorMsg);
+
+		} while (errorMsg != null);
+
+		do {
+
+			m_email = Console.read("Email > ");
+			errorMsg = Check.emailUnique(m_email);
+
+			if (errorMsg != null)
+				Console.write(errorMsg);
+
+		} while (errorMsg != null);
+
+		do {
+
+			m_password = Console.read("Password > ");
+			errorMsg = Check.password(m_password);
+
+			if (errorMsg != null)
+				Console.write(errorMsg);
+
+		} while (errorMsg != null);
+
+		do {
+
+			m_SSID = Console.read("SSID > ");
+			errorMsg = Check.ssidUnique(m_SSID);
+
```

Comment:
Just use the `Check.ssid()` method, because in a few days it will be the same as `Check.ssidUnique()` + this last one will be removed

Reply:
Fixed